tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type-casting.bal

```diff
@@ -27,6 +28,133 @@ function stringtoint(string value) returns int|error {
     return result;
 }
 
+function testByteArrayToIntArray() {
```

Comment:
Shall we add one test case covering all cases. Separate functions are not clear as well.

Reply:
```
function testIntArrayCasting() {
    byte[] byteArray = [1, 128, 255];
    int:Signed8[] signed8Array = [-128, 0, 127];
    int:Signed16[] signed16Array = [-128, 0, 127];
    int:Signed32[] signed32Array = [-128, 0, 127];
    int:Unsigned8[] unsigned8Array = [5, 55, 255];
    int:Unsigned16[] unsigned16Array = [5, 5050, 65535];
    int:Unsigned32[] unsigned32Array = [50, 5050, 655355];

    any anyByteArray = byteArray;
    any anySigned8Array = signed8Array;
    any anySigned16Array = signed16Array;
    any anySigned32Array = signed32Array;
    any anyUnsigned8Array = unsigned8Array;
    any anyUsigned16Array = unsigned16Array;
    any anyUsigned32Array = unsigned32Array;

    //byte
    test:assertEquals(<int[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Signed16[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Signed16[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Signed32[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Unsigned8[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Unsigned16[]>anyByteArray, [1, 128, 255]);
    test:assertEquals(<int:Unsigned32[]>anyByteArray, [1, 128, 255]);

    //Signed8
    test:assertEquals(<int[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Signed16[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Signed16[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Signed32[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Unsigned8[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Unsigned16[]>anySigned8Array, [1, 128, 255]);
    test:assertEquals(<int:Unsigned32[]>anySigned8Array, [1, 128, 255]);

    // Signed16, Signed32,Unsigned8, Unsigned16, Unsigned32
}
```

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31302/commits/991078ac38c0c57202f4c1fac46b53e70911080a